### DISCUSSION TITLE
Specs for new color schemes

### DISCUSSION THREAD

**Discussion by vefatica** [microsoft/terminal#17303] [Q&A] [answered]
Can you guide me to a file containing the specifications (16-color palette, foreground, background, cursor) for the 3 new color schemes?

Thanks!

**Comment by DHowett** [comment, marked as answer]
No formal specifications exist for them. They were discussed in...

* Dark+: https://github.com/microsoft/terminal/issues/6176
* IBM and CGA: https://github.com/microsoft/terminal/issues/16754, https://github.com/microsoft/terminal/issues/16754#issuecomment-2024066880 (specifically)

**Comment by vefatica** [comment]
Thanks.  Are they in any source code that's available to me?  Oh, I just discovered that I can dig them out of the settings dialog (one color at a time)?  I have a lot of time; that works for me.

Can the selection colors be set with a control sequence?

**Comment by vefatica** [comment]
Oops!  No I can't dig them out of settings.  That was just for the "Billow" scheme (which is spec'd in my settings file).  Where did that come from?

**Reply by DHowett** [reply]
Uhh... that's not one of ours! It probably came from a Fragment Extension. Check out `%LOCALAPPDATA%\Microsoft\Windows Terminal\Fragments` (user) and `%PROGRAMDATA%\Microsoft\Windows Terminal\Fragments` (system).

If it's not in there, it came from an app package and those are harder to find.

Honestly, the easiest way to dig out any scheme will be to [Create a copy], [Save], and then get it out of _your_ settings file.

**Comment by vefatica** [comment]
How about setting the selection colors with a control sequence ... possible?

**Comment by vefatica** [comment]
I think something is amiss.  I made a copy of "CGA" and got this (below) in my settings file.  I don't think the cursor is supposed to be green! 

```
        {
            "background": "#000000",
            "black": "#000000",
            "blue": "#0000AA",
            "brightBlack": "#555555",
            "brightBlue": "#5555FF",
            "brightCyan": "#55FFFF",
            "brightGreen": "#55FF55",
            "brightPurple": "#FF55FF",
            "brightRed": "#FF5555",
            "brightWhite": "#FFFFFF",
            "brightYellow": "#FFFF55",
            "cursorColor": "#00AA00",
            "cyan": "#00AAAA",
            "foreground": "#AAAAAA",
            "green": "#00AA00",
            "name": "CGA (Copy)",
            "purple": "#AA00AA",
            "red": "#AA0000",
            "selectionBackground": "#FFFFFF",
            "white": "#AAAAAA",
            "yellow": "#AA5500"
        }
```

It's actually not green if I use CGA or CGA (Copy) for a profile but it is green (no surprise) if I manually (control sequences) set the colors to the above (as saved in my settings file).
.
![image](https://github.com/microsoft/terminal/assets/61856645/54e026aa-740a-4c27-b2af-6fbe3240d93a)

**Comment by vefatica** [comment]
The same thing happens when I copy "IBM 5153".  The settings file's selection background looks wrong too; like the cursor color, it's not what you get if you assign the theme to a profile.

![image](https://github.com/microsoft/terminal/assets/61856645/37f162bd-2225-405f-adc3-6f0de1ccc5b4)

**Comment by DHowett** [comment]
> It's actually not green if I use CGA or CGA (Copy) for a profile

???

This is what I get when I set a profile to use CGA.

![image](https://github.com/microsoft/terminal/assets/189190/a790246d-bed4-4c6b-b258-acfa2a94d340)

It absolutely is for me. If I had to guess, something about your shell profile is overwriting it.

> Selection background

The selection color is always blended at 50% or 80% opacity (I can't remember which) with the rest of the terminal. It's a bit of a misnomer to call it a background, or to say it is a solid color.

**Comment by vefatica** [comment]
It's not my shell.  Here's CMD, configured to use the "CGA" scheme.

![image](https://github.com/microsoft/terminal/assets/61856645/1e22c40b-82ee-4188-8281-6855f4b45a85)

And settings also shows a white cursor.

![image](https://github.com/microsoft/terminal/assets/61856645/b3e1fb3c-e9dd-4257-a110-453d0216585f)

I'm using WindowsTerminal 1.21.2405.06002 (preview).

**Comment by vefatica** [comment]
OK, I had this (below).  If I did that it was YEARS ago.  It was overriding (wrongly, as I see it) the scheme-specified value.  What about selectionBackground?  CGA Copy says FFFFFF but that's not what I get.

```
    "profiles": 
    {
        "defaults": 
        {
            "closeOnExit": "always",
            "colorScheme": "Campbell",
            "cursorColor": "#C0C0C0",
```

**Reply by DHowett** [reply]
Can you show me what you _do_ get?

**Comment by vefatica** [comment]
For CGA selectionBackground (also CGA copy) I get this.  It's 0x808080.

![image](https://github.com/microsoft/terminal/assets/61856645/4dc50015-f311-47ac-b9d3-8466ac7a0573)

**Reply by DHowett** [reply]
> The selection color is always blended at 50% or 80% opacity (I can't remember which) with the rest of the terminal. It's a bit of a misnomer to call it a background, or to say it is a solid color.

Guess that means it’s 50%!

FFFFFF x 0.50 + 000000 = 808080

**Reply by DHowett** [reply]
Funny enough, I just stumbled across https://github.com/microsoft/terminal/issues/11181 . . . which you filed, and where we discussed the selection color blending. Time is a flat circle!

**Comment by vefatica** [comment]
OK, thanks for the explanation.  As you can see, the cursor is green now that I got rid of that "default" value.

**Reply by DHowett** [reply]
Excellent!
